crates/bevy_ecs/src/reflect/entity_commands.rs

```diff
@@ -200,10 +204,14 @@ fn insert_reflect(
     let Some(type_registration) = type_registry.get(type_info.type_id()) else {
         panic!("Could not get type registration (for component type {type_path}) because it doesn't exist in the TypeRegistry.");
     };
-    let Some(reflect_component) = type_registration.data::<ReflectComponent>() else {
-        panic!("Could not get ReflectComponent data (for component type {type_path}) because it doesn't exist in this TypeRegistration.");
-    };
-    reflect_component.insert(&mut entity, component.as_partial_reflect(), type_registry);
+
+    if let Some(reflect_bundle) = type_registration.data::<ReflectBundle>() {
+        reflect_bundle.insert(&mut entity, component.as_partial_reflect(), type_registry);
+    } else if let Some(reflect_component) = type_registration.data::<ReflectComponent>() {
```

Comment:
Minor nit: I suspect components to be used more often here than bundles. Could we swap the order of these if-statements to optimize for the component case?

Reply:
sure! i got it